src/rpc/ngHandlers/LedgerData.cpp

```diff
@@ -0,0 +1,221 @@
+{
+    // marker must be int if outOfOrder is true
+    if (input.outOfOrder && input.marker)
+        return Error{RPC::Status{RPC::RippledError::rpcINVALID_PARAMS, "outOfOrderMarkerNotInt"}};
+    if (!input.outOfOrder && input.diffMarker)
+        return Error{RPC::Status{RPC::RippledError::rpcINVALID_PARAMS, "markerNotString"}};
+
+    auto const range = sharedPtrBackend_->fetchLedgerRange();
+    auto const lgrInfoOrStatus = RPC::getLedgerInfoFromHashOrSeq(
+        *sharedPtrBackend_, ctx.yield, input.ledgerHash, input.ledgerIndex, range->maxSequence);
+
+    if (auto const status = std::get_if<RPC::Status>(&lgrInfoOrStatus))
+        return Error{*status};
+
+    auto const lgrInfo = std::get<ripple::LedgerInfo>(lgrInfoOrStatus);
+
+    Output output;
+    // no marker -> first call, return header information
+    auto header = boost::json::object();
+    if ((!input.marker) && (!input.diffMarker))
+    {
+        if (input.binary)
+        {
+            header[JS(ledger_data)] = ripple::strHex(RPC::ledgerInfoToBlob(lgrInfo));
+        }
+        else
+        {
+            header[JS(accepted)] = true;
+            header[JS(account_hash)] = ripple::strHex(lgrInfo.accountHash);
+            header[JS(close_flags)] = lgrInfo.closeFlags;
+            header[JS(close_time)] = lgrInfo.closeTime.time_since_epoch().count();
+            header[JS(close_time_human)] = ripple::to_string(lgrInfo.closeTime);
+            header[JS(close_time_resolution)] = lgrInfo.closeTimeResolution.count();
+            header[JS(hash)] = ripple::strHex(lgrInfo.hash);
+            header[JS(ledger_hash)] = ripple::strHex(lgrInfo.hash);
+            header[JS(ledger_index)] = std::to_string(lgrInfo.seq);
+            header[JS(parent_close_time)] = lgrInfo.parentCloseTime.time_since_epoch().count();
+            header[JS(parent_hash)] = ripple::strHex(lgrInfo.parentHash);
+            header[JS(seqNum)] = std::to_string(lgrInfo.seq);
+            header[JS(totalCoins)] = ripple::to_string(lgrInfo.drops);
+            header[JS(total_coins)] = ripple::to_string(lgrInfo.drops);
+            header[JS(transaction_hash)] = ripple::strHex(lgrInfo.txHash);
+        }
+        header[JS(closed)] = true;
+        output.header = std::move(header);
+    }
+    else
+    {
+        if (input.marker && !sharedPtrBackend_->fetchLedgerObject(*(input.marker), lgrInfo.seq, ctx.yield))
+            return Error{RPC::Status{RPC::RippledError::rpcINVALID_PARAMS, "markerDoesNotExist"}};
+    }
+
+    output.ledgerHash = ripple::strHex(lgrInfo.hash);
+    output.ledgerIndex = lgrInfo.seq;
+
+    auto const start = std::chrono::system_clock::now();
+    std::vector<Backend::LedgerObject> results;
+    if (input.diffMarker)
+    {
+        // keep the same logic as previous implementation
+        auto const diff = sharedPtrBackend_->fetchLedgerDiff(*(input.diffMarker), ctx.yield);
+        std::vector<ripple::uint256> keys;
+        for (auto& [key, object] : diff)
```

Comment:
Old handler uses " for (auto&& [key, object] : diff)" . change to auto&, auto && is confusing here

Reply:
indeed . it should not be const in this case. will fix